smartpy/core/trainer.py

```diff
@@ -0,0 +1,85 @@
+from collections import OrderedDict
+
+import theano
+from time import time
+from .status import Status
+
+
+class Trainer(object):
+    def __init__(self, optimizer, status=None):
```

Comment:
Should have a StoppingCriterion in the constructor. Training forever is not a really valid state for the trainer. It is valid only for online learning but I think it's better if it is explicit like MaxEpoch(np.inf).